docs/contract-interface/serialization-interface.md

```diff
@@ -4,8 +4,43 @@ sidebar_position: 5
 
 # Serialization Protocols
 
-TODO
+Serialization formats within the SDK define how data structures are translated into bytes which are needed for passing data into methods of the smart contract or storing data in state. For the case of method parameters, [JSON](https://www.json.org/json-en.html) (default) and [Borsh](https://borsh.io/) are supported with the SDK and for storing data on-chain Borsh is used.
 
-<!-- JSON vs Borsh and when to use either -->
+The qualities of JSON and Borsh are as follows:
 
-<!-- Description of json wrapper types like U128 -->
\ No newline at end of file
+JSON:
+- Human readable
+- Self-describing format (don't need to know the underlying type)
+- Easy interop with JavaScript
+- Less efficient size and (de)serialization
+
+Borsh:
+- Compact, binary format that's efficient for serialized data size
+- Need to know data format or have a schema to deserialize data
+- Strict and canonical binary representation
+- Fast and less overhead in most cases
+
+In general, JSON will be used for contract calls and cross-contract calls for a better DevX, where Borsh can be used to optimize using less gas by having smaller parameter serialization and less deserialization computation within the contract.
+
+### Overriding Serialization Protocol Default
+
+The result and parameter serialization can be opted into separately, but all parameters must be of the same format (can't serialize some parameters as borsh and others as JSON). An example of switching both the result and parameters to borsh is as follows:
+
+```rust
+#[result_serializer(borsh)]
+pub fn sum_borsh(#[serializer(borsh)] a: u32, #[serializer(borsh)] b: u32) -> u32 {
+    a + b
+}
+```
+
+Where the `result_serializer(borsh)` annotation will override the default result serialization protocol from JSON to borsh and the `serializer(borsh)` annotations will override the parameter serialization.
```

Comment:
Cool, learning new stuff. So we don't need to apply an annotation to the return value type?